irods/test/data_obj_test.py

```diff
@@ -2362,6 +2363,79 @@ def _update(n):
 
         self.assertEqual(pbar.percent_done(), 100)
 
+    def test_replica_truncate_related_errors__issue_534(self):
+        sess = self.sess
+        data_objs = self.sess.data_objects
+        truncated_size = 16
+
+        try:
+            # Set path for a new, as yet nonexisting data object.
+            data_path = '{}/{}'.format(
+                helpers.home_collection(sess),
+                unique_name(my_function_name(), datetime.now(), truncated_size) + "issue_534_errors")
+            self.assertFalse (data_objs.exists(data_path), "Data object should not yet exist.")
+
+            # Assert appropriate error is raised for the nonexisting object.
+            with self.assertRaises(ex.OBJ_PATH_DOES_NOT_EXIST):
+                data_objs.replica_truncate(data_path, truncated_size)
+
+            d = data_objs.create(data_path)
+
+            # Assert appropriate error is raised for an existing object without a replica on the given resource.
+            with self.create_simple_resc() as newResc:
+                with self.assertRaises(ex.SYS_REPLICA_INACCESSIBLE):
+                    d.replica_truncate(truncated_size, **{kw.RESC_NAME_KW: newResc})
+
+            # Assert appropriate error is raised for an invalid (negative) size argument.
+            try:
+                data_objs.replica_truncate(data_path, -truncated_size)
+            except Exception as e:
+                self.assertIsInstance(e, ex.UNIX_FILE_TRUNCATE_ERR)
+                self.assertEqual('EINVAL', errno.errorcode[abs(e.code) % 1000])
+
+        finally:
+            if data_objs.exists(data_path):
+                data_objs.unlink(data_path, force = True)
+
+    def test_replica_truncate__issue_534(self):
+        sess = self.sess
+        data_objs = self.sess.data_objects
+        original_size = 16
+        original_content = b'_' * original_size
+        with self.create_simple_resc() as newResc:
+            for truncated_size in (original_size//2, original_size + 1024):
+                try:
+                    data_path = '{}/{}'.format(
+                        helpers.home_collection(sess),
+                        unique_name(my_function_name(), datetime.now(), truncated_size) + "_issue_534")
+
+                    # Create a data object with size original_size.
+                    with data_objs.open(data_path,'w') as f:
+                        f.write(original_content)
+                    d = data_objs.get(data_path)
+
+                    # Ensure there are two replicas, one on newResc as well as one on 'demoResc'
+                    d.replicate(resource = newResc)
+                    d.replica_truncate(truncated_size, **{kw.RESC_NAME_KW: newResc})
```

Comment:
Can we capture the JSON output here and perhaps assert that the replica number and resource hierarchy are as expected?

Reply:
It appears the "boilerplate" code typically used to return a response does not return any embedded JSON.  It might be accessible in some future version of the PRC, if processing / testing of such return messages is  useful in the client.

Reply:
So... I guess I'm confused, then. What does this line in `data_object_manager.replica_truncate` do (line 566):
```python
        return json.loads(msg.myStr)
```
It looks to me like this method returns a JSON object.

Reply:
I was looking at the wrong code, sorry! You are correct.

Reply:
Ok we're doing that now.  Should I also assert msg is empty for the routine ( non-error )case?

Reply:
Check. ok, resolving